### PR TITLE
Fix connection for doctrine event tag

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

After #355 I got some errors like:
```
The Doctrine connection "%simplethings.entityaudit.connection%" referenced   
!!    in service "simplethings_entityaudit.log_revisions_listener" does not exist  
!!    . Available connections names: "default".   
```

It look like doctrine event tag will not convert parameter name into value. This PR will replace this parameter name into value.

I am targeting this branch, because this change respect BC.

Changelog is not needed becouse we do not release #355 yet.

### CODE REVIEW

**Comment by wbloszyk** [comment]
@phansys 
I check in my project and it is working. Anyway I can not fix tests. Can you look at this?

**Comment by phansys** [comment]
> @phansys
> I check in my project and it is working. Anyway I can not fix tests. Can you look at this?

Sure, I'll try to check this later.

**Comment by phansys** [comment]
I think this is working, at least in the test environment: wbloszyk/EntityAuditBundle#1.
Please, confirm if you can check these changes in a working project.

**Comment by wbloszyk** [comment]
> I think this is working, at least in the test environment: [wbloszyk#1](https://github.com/wbloszyk/EntityAuditBundle/pull/1).
> Please, confirm if you can check these changes in a working project.

@phansys 
Parameter will be always the same `simplethings.entityaudit.connection`. We do not have to check it, becouse it is set in `load()`.

File: src/DependencyInjection/SimpleThingsEntityAuditExtension.php [hunk 1]
```diff
@@ -49,5 +49,26 @@ public function load(array $configs, ContainerBuilder $container): void
...
+    }
+
+    private function fixParametersFromDoctrineEventSubscriberTag(ContainerBuilder $container, array $definitionNames): void
+    {
+        foreach ($definitionNames as $definitionName) {
+            $definition = $container->getDefinition($definitionName);
+            $tags = $definition->getTag('doctrine.event_subscriber');
+            $definition->clearTag('doctrine.event_subscriber');
+
+            foreach ($tags as $id => $attributes) {
```
Comment (phansys) [inline src/DependencyInjection/SimpleThingsEntityAuditExtension.php:66]:
```suggestion
            foreach ($tags as $attributes) {
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by phansys — APPROVED** [review]

File: src/DependencyInjection/SimpleThingsEntityAuditExtension.php [hunk 2]
```diff
@@ -49,5 +49,26 @@ public function load(array $configs, ContainerBuilder $container): void
...
+    private function fixParametersFromDoctrineEventSubscriberTag(ContainerBuilder $container, array $definitionNames): void
+    {
+        foreach ($definitionNames as $definitionName) {
+            $definition = $container->getDefinition($definitionName);
+            $tags = $definition->getTag('doctrine.event_subscriber');
+            $definition->clearTag('doctrine.event_subscriber');
+
+            foreach ($tags as $attributes) {
+                if (isset($attributes['connection'])) {
+                    $attributes['connection'] = (string) $container->getParameter('simplethings.entityaudit.connection');
```
Comment (phansys) [inline src/DependencyInjection/SimpleThingsEntityAuditExtension.php:68]: IMO, if we harcode the parameter name here, we are making less flexible the replacing mechanism.
If in the future the parameter name used in these tags is renamed, we must replace this occurrence too.
I also think that the final users could not override these services if they need to use a different value.

Comment (wbloszyk) [reply src/DependencyInjection/SimpleThingsEntityAuditExtension.php:68]: We set this `simplethings.entityaudit.connection` parameter in the same `SimpleThingsEntiyAuditExtension`, so I think we do not have to afriaid about rename. Additionaly if users override this service then connection name must be set directly, oherwise expection will be throw. 

We can also move it to the compiler pass. Then we can check whole services tagged by `doctrine.event_listener` and replace `simplethings.entityaudit.connection` parameter in `connetion` to propertly value.

Comment (wbloszyk) [reply src/DependencyInjection/SimpleThingsEntityAuditExtension.php:68]: https://github.com/symfony/symfony/pull/40262
WDYT? @sonata-project/contributors

Comment (VincentLanglet) [reply src/DependencyInjection/SimpleThingsEntityAuditExtension.php:68]: I'm not sure to understand, should we wait for https://github.com/symfony/symfony/pull/40262 or can we merge this ?

Comment (wbloszyk) [reply src/DependencyInjection/SimpleThingsEntityAuditExtension.php:68]: We can merge this now. This code allow use param only for ower services.we can remove it when symfony accept it and we bump doctrine-bridge.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**